third_party/move/move-vm/runtime/src/native_functions.rs

```diff
@@ -81,8 +81,26 @@ impl NativeFunctions {
     where
         I: IntoIterator<Item = (AccountAddress, Identifier, Identifier, NativeFunction)>,
     {
+        let vector_bytecode_instruction_methods = HashSet::from([
+            "empty",
+            "length",
+            "borrow",
+            "borrow_mut",
+            "push_back",
+            "pop_back",
+            "destroy_empty",
+            "swap",
+        ]);
+
         let mut map = HashMap::new();
         for (addr, module_name, func_name, func) in natives.into_iter() {
+            if module_name.as_str() == "string"
+                && vector_bytecode_instruction_methods.contains(func_name.as_str())
+            {
+                println!("ERROR: Tried to register as native a vector bytecode_instruction method {}, skipping.", func_name.as_str());
+                continue;
```

Comment:
not sure if this check is even needed (this is mostly to make sure we don't create new native function with same name as bytecode instruction), or if it's better to remove it - or on the other hand `panic` here.

but this is basically to be a fixed replacement of the filter in `aptos-move/aptos-vm/src/natives.rs`